### PR TITLE
Append slash to log folder path string, if it doesnt already contain a slash in C++ SDK

### LINKED ISSUE #78: The C++ SDK is writing its log file to the wrong folder
**Describe the bug**
The C++ SDK is writing its log file to the wrong folder. The log file is not in the folder specified, but next to the folder

**To Reproduce**
1. Log a message with the log functionality of the C++ SDK
2. Observe that the log file is not in the folder specified, but next to the folder

**Expected behavior**
the log file should be inside that folder not next to it.

**Screenshots**
Error

**Desktop (please complete the following information):**
 - C++ SDK

**Fix idea**
Either add a slash [here](https://github.com/PlayFab/gsdk/blob/master/cpp/cppsdk/gsdk.cpp#L158) or the config value LOG_FOLDER_KEY ("logFolder") needs to get a slash appended in the end

### PR DESCRIPTION
**Pull request by stkrwork**
**What this PR does / why we need it**:
closes #78. Adds a slash to the log folder path in the C++ SDK, so that the log file gets written to the correct folder in process based mode.

### CODE REVIEW

**Comment by dgkanatsios** [comment]
hey @stkrwork, thanks again for the great work, we'll review soon! I assume this will work both on Windows and Linux?

File: cpp/cppsdk/gsdk.cpp [hunk 1]
```diff
@@ -155,6 +155,11 @@ namespace Microsoft
                     logFolder = "";
                 }
 
+                if (logFolder.back() != '/' && logFolder.back() != '\\')
+                {
+                    logFolder.append("/");
```
Comment (pgilmorepf) [inline cpp/cppsdk/gsdk.cpp:160]:
```suggestion
                    logFolder.append(System.IO.Path.DirectorySeparatorChar);
```

File: cpp/cppsdk/gsdk.cpp [hunk 2]
```diff
@@ -155,6 +155,11 @@ namespace Microsoft
                     logFolder = "";
                 }
 
+                if (logFolder.back() != '/' && logFolder.back() != '\\')
```
Comment (pgilmorepf) [inline cpp/cppsdk/gsdk.cpp:158]:
```suggestion
                if (!logFolder.EndsWith(System.IO.Path.DirectorySeparatorChar) && !logFolder.EndsWith(System.IO.Path.AltDirectorySeparatorChar))
```

**Review by pgilmorepf — CHANGES_REQUESTED** [review]
Please apply the suggested changes, and re-test.

File: cpp/cppsdk/gsdk.cpp (hunk 2 above)
Comment (stkrwork) [reply cpp/cppsdk/gsdk.cpp:158]: std::string doesn't offer a function called "EndsWith(...)"

File: cpp/cppsdk/gsdk.cpp (hunk 1 above)
Comment (stkrwork) [reply cpp/cppsdk/gsdk.cpp:160]: System.IO.Path.DirectorySeparatorChar is part of .Net Framework, which is not useable in C++

File: cpp/cppsdk/gsdk.cpp (hunk 2 above)
Comment (pgilmorepf) [reply cpp/cppsdk/gsdk.cpp:158]: Ugh, you're right, it's 5.0+.
https://docs.microsoft.com/en-us/dotnet/api/system.string.endswith?view=net-5.0
Let's just switch the slashes to System.IO.Path.DirectorySeparatorChar then. Thanks & Sorry.

**Review by pgilmorepf — APPROVED** [review]

File: cpp/cppsdk/gsdk.cpp [hunk 3]
```diff
@@ -154,7 +154,17 @@ namespace Microsoft
                 {
                     logFolder = "";
                 }
-
+#if _WIN32
```
Comment (dgkanatsios) [inline cpp/cppsdk/gsdk.cpp:157]: should this be just `#if WIN32`, similar to that? https://github.com/PlayFab/gsdk/blob/a329d1ba08a26ef8a122462212f10d98b0bd62c8/cpp/cppsdk/include/playfab/PlayFabBaseModel.h#L14

Comment (dgkanatsios) [reply cpp/cppsdk/gsdk.cpp:157]: it's also mentioned here as WIN32
https://github.com/PlayFab/gsdk/blob/master/cpp/CMakeLists.txt#L10

Comment (stkrwork) [reply cpp/cppsdk/gsdk.cpp:157]: upon some research _WIN32 is always defined, and WIN32 can also be not defined if no windows header is included, so I would stick with _WIN32

Comment (dgkanatsios) [reply cpp/cppsdk/gsdk.cpp:157]: nice, confirming that here! https://stackoverflow.com/a/662095 
Thanks!

**Comment by dgkanatsios** [comment]
merged, thanks for the contribution!

**Pull request merged**